modules/functions/GeoGridFunction.cc

```diff
@@ -212,24 +218,211 @@ bool GeoGridFunction::canOperateOn(DDS &dds)
     bool usable = false;
 
     // Go find all the Grid variables.
-	//vector<Grid *> *grids = new vector<Grid *>();
-	vector<Grid*> grids;
-	get_grids(dds, &grids);
+    //vector<Grid *> *grids = new vector<Grid *>();
+    vector<Grid*> grids;
+    get_grids(dds, &grids);
 
-	// Were there any?
+    // Were there any?
     if(!grids.empty()){
-    	// Apparently so...
+        // Apparently so...
 
-    	// See if any one of them looks like suitable GeoGrid
-    	vector<Grid *>::iterator git;
-    	for(git=grids.begin(); !usable && git!=grids.end() ; git++){
-    		Grid *grid = *git;
-    		usable = is_geo_grid(grid);
-    	}
+        // See if any one of them looks like suitable GeoGrid
+        vector<Grid *>::iterator git;
+        for(git=grids.begin(); !usable && git!=grids.end() ; git++){
+            Grid *grid = *git;
```

Comment:
This is a place - the '.begin()' and '.end()' methods are a hint - that the new range-based for-loop can be used. This can be rewritten:

for (auto grid: grids) {
    if (is_geo_grid(grid) {
         return true;
    }
}

No need to make the change this time, but the pattern can save time,